### PR TITLE
search: improve filter configuration

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Replaces the button by a toggle on the filter display.
* Adds URL parameters on searchFilter.
* Updates the routing configuration to show all options.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>
Co-Authored-by: Sébastien Délèze <sebastien.deleze@rero.ch>

## Why are you opening this PR?

- Improve search filters to add some options

## How to test?

- Check configuration and test filters on `Backend` url

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/ng-core-tester/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -309,9 +309,33 @@ const routes: Routes = [
...
+              filter: 'simple',
+              value: '0',
+              /* If you set this value, the url parameter will still be present,
+              but with different values */
+              disabledValue: '1',
+              url: {
+                routerLink: ['/record', 'search', 'documents'],
+                title: 'Link to search document'
+              }
+            },
```
Comment (zannkukai) [inline projects/ng-core-tester/src/app/app-routing.module.ts:323]: If I understand : an advanced search url should be : `http://localhost/record/search/documents?simple=1` ?
For me, this url seems strange :(

File: projects/rero/ng-core/src/lib/record/record.ts [hunk 2]
```diff
@@ -66,5 +66,15 @@ export interface SearchField {
 export interface SearchFilter {
   filter: string;
   label: string;
-  value?: string;
+  value: string;
+  /* If you set this value, the url parameter will still be present,
+  but with different values */
+  disabledValue?: string;
```
Comment (zannkukai) [inline projects/rero/ng-core/src/lib/record/record.ts:72]: Maybe we could use a boolean. If false, then the url parameter will be present but whitout any value (seems valid for URL's). Ex : http://localhost/?param1=stuff&param2

File: projects/ng-core-tester/src/app/app-routing.module.ts (hunk 1 above)
Comment (Garfield-fr) [reply projects/ng-core-tester/src/app/app-routing.module.ts:323]: advanced search url is `http://localhost/record/search/documents?simple=0`

**Review by iGormilhit — APPROVED** [review]

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts [hunk 3]
```diff
@@ -891,4 +915,20 @@ export class RecordSearchComponent implements OnInit, OnChanges, OnDestroy {
       ? this._config.aggregationsName[key]
       : null;
   }
+
+  /** Extract persistent search filters on current url */
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.ts:919]: Just add @returns in the doc block.

**Review by sebastiendeleze — APPROVED** [review]

**Pull request merged**